symengine/mp_class.h

```diff
@@ -276,8 +298,8 @@ inline auto get_mpz_t(const piranha::integer &i) -> decltype(i.get_mpz_view())
     return i.get_mpz_view();
 }
 
-inline void mp_pow_ui(piranha::integer &res, const piranha::integer &i,
-                      unsigned long n)
+template <typename T>
+inline void mp_pow_ui(T &res, const T &i, unsigned long n)
```

Comment:
Can you repeat this function without using templates so that only `piranha::integer` and `piranha::rational` can be used?